modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/util/CommerceAccountRoleHelperImpl.java

```diff
@@ -231,6 +235,23 @@ else if (name.equals(AccountRoleConstants.ROLE_NAME_SUPPLIER)) {
 					CommercePriceListActionKeys.ADD_COMMERCE_PRICE_LIST
 				});
 
+			companyResourceActionIds.put(
```

Comment:
when adding these kind of permission just be sure that we are not allowing the supplier to do it also on entities outside of it's scope, like on products that are in catalogs that are not linked to him
this should be verified using headless APIs

Reply:
Checked the API calls for these and they are good. Only one I was not able to confirm is
`CommercePricingClassActionKeys.ADD_COMMERCE_PRICING_CLASS`
but I dont think there is any API for that.